src/orchestration/orchestration.cpp

```diff
@@ -616,13 +620,95 @@ Ref<OScriptFunction> Orchestration::create_function(const MethodInfo& p_method,
     function->_method = p_method;
     function->_owning_node_id = p_node_id;
     function->_user_defined = p_user_defined;
+    function->_returns_value = MethodUtils::has_return_value(p_method) ;
 
     _functions[p_method.name] = function;
 
     _self->emit_signal("functions_changed");
 
     return function;
 }
+Ref<OScriptFunction> Orchestration::duplicate_function(const StringName& p_name, bool p_include_code)
+{
+    ERR_FAIL_COND_V_MSG(_has_instances(), nullptr, "Cannot duplicate functions, instances exist.");
+    ERR_FAIL_COND_V_MSG(!has_function(p_name), nullptr, "No function exists with the name: " + p_name);
+
+    Ref<OScriptGraph> old_graph = find_graph(p_name);
+    Ref<OScriptFunction> old_function = find_function(p_name);
+    Ref<OScriptFunction> result;
+
+    // make a unique name for the new function
+    String new_name = NameUtils::create_unique_name(p_name, get_function_names());
+
+    // make a graph
+    Ref<OScriptGraph> graph = create_graph(new_name, OScriptGraph::GF_FUNCTION | OScriptGraph::GF_DEFAULT);
+
+
+    // duplicate each node, make a dictionary that maps old node IDs to new node IDs
+    Dictionary node_id_map;
```

Comment:
I would suggest using `HashMap<int, int>` rather than a `Dictionary`. It will bypass the coercion into a Variant and will perform better for larger graphs that get copied.